### DISCUSSION TITLE
`User` gets different roles and permissions depending on `tenant`

### DISCUSSION THREAD

**Discussion by bilogic** [spatie/laravel-permission#2775] [Q&A]
I see github as a same domain, multitenant platform. e.g. everyone has an account and their own repositories (`owner`), but I may not be a `collaborator` of repositories that do not belong to me and so I have different permissions depending on repository (tenant).

So, some of the questions I have are:

1. In this tenant (repo), what are the authenticated user's role(s)?
2. In this tenant (repo), what are the authenticated user's permissions?

Is there a short readable way to code this up with laravel-permission?

**Comment by drbyte** [comment]
The package's usual API still applies, but you'll need to implement the teams feature.
Then once you've selected the user and set the team ID, you can query the user's roles and directly-assigned permissions as usual

**Reply by bilogic** [reply]
Thanks for getting back, going further with the github example,

1. For normal users, I see each repository as a `team`
2. But suppose staff of github also needs to login for administrative purposes, they need to be assigned a team as well, correct? 
3. My concern is that this github staff `team` will not be a repository, but on my end, it seems I have to keep them in the same `team` index space, is this how teams was designed to be used?
4. And how should I handle someone who is both a staff and also has repositories? i.e. which team do I set them as?
5. I checked that `setPermissionsTeamId()` accepts `string`, and thought that perhaps `repo:1` and `staff` would work but that led to an exception because `model_has_roles` expected `team_id` to be an integer

I think addressing my 4th question would address everything else

After some thoughts, why doesn't a `Team` model exists? That would make it possible to write staff `belongsTo` a team and repo `belongsTo` a team.
